### PR TITLE
feat(builder): apply css-minimizer-webpack-plugin when use rspack

### PR DESCRIPTION
**Pull request by 9aoy**
### <samp>🤖 Generated by Copilot at f73abe7</samp>

This pull request improves the CSS minification feature of the builder package by using a shared CssMinimizerPlugin for both webpack and rspack bundlers. It also updates the tests, the documentation, and the dependencies to reflect this change. Additionally, it enhances the JavaScript minification feature of the rspack bundler by using the SwcJsMinimizerPlugin.

### <samp>🤖 Generated by Copilot at f73abe7</samp>

*  Add a changeset file to document the changes and version bumps for the builder packages ([link](https://github.com/web-infra-dev/modern.js/pull/4602/files?diff=unified&w=0#diff-191424d7151649626b2d163415f957d66d6ee9c356a986c5a4d113d4341760f4R1-R9))
*  Refactor the minimize plugin for the builder-rspack-provider package to use the bundler chain instead of the rspack config object, and to reuse the CssMinimizerPlugin from the builder-shared package ([link](https://github.com/web-infra-dev/modern.js/pull/4602/files?diff=unified&w=0#diff-00652a3b9cfcf087a07f68d8d17fcba9f6529ef9ad06eff43a5d63bb7be474f3L1-R11), [link](https://github.com/web-infra-dev/modern.js/pull/4602/files?diff=unified&w=0#diff-00652a3b9cfcf087a07f68d8d17fcba9f6529ef9ad06eff43a5d63bb7be474f3L42-R46), [link](https://github.com/web-infra-dev/modern.js/pull/4602/files?diff=unified&w=0#diff-00652a3b9cfcf087a07f68d8d17fcba9f6529ef9ad06eff43a5d63bb7be474f3L49-R62))
*  Update the test cases for the minimize plugin for the builder-rspack-provider package to match the new bundler chain configuration and snapshots ([link](https://github.com/web-infra-dev/modern.js/pull/4602/files?diff=unified&w=0#diff-996b6ead70a814bcbf609de19b0dd583899f30f5558e5c0bc8f6dd641145f79dL18-R18), [link](https://github.com/web-infra-dev/modern.js/pull/4602/files?diff=unified&w=0#diff-996b6ead70a814bcbf609de19b0dd583899f30f5558e5c0bc8f6dd641145f79dL36-R74), [link](https://github.com/web-infra-dev/modern.js/pull/4602/files?diff=unified&w=0#diff-996b6ead70a814bcbf609de19b0dd583899f30f5558e5c0bc8f6dd641145f79dL83-R154), [link](https://github.com/web-infra-dev/modern.js/pull/4602/files?diff=unified&w=0#diff-996b6ead70a814bcbf609de19b0dd583899f30f5558e5c0bc8f6dd641145f79dL110-R216), [link](https://github.com/web-infra-dev/modern.js/pull/4602/files?diff=unified&w=0#diff-996b6ead70a814bcbf609de19b0dd583899f30f5558e5c0bc8f6dd641145f79dL140-R275))
*  Add the css-minimizer-webpack-plugin dependency to the builder-shared package, and export the applyCSSMinimizer function to apply the CssMinimizerPlugin to any bundler chain ([link](https://github.com/web-infra-dev/modern.js/pull/4602/files?diff=unified&w=0#diff-fddf5e8ef6a1c41374dc69b604fbc5fbd1bc944f335e853797d9badbd5455fa9R158), [link](https://github.com/web-infra-dev/modern.js/pull/4602/files?diff=unified&w=0#diff-653fc21c778ad910cf02b8c68af4da264f5d070ebb9811c8c39f693f6486caf5L1-R7), [link](https://github.com/web-infra-dev/modern.js/pull/4602/files?diff=unified&w=0#diff-653fc21c778ad910cf02b8c68af4da264f5d070ebb9811c8c39f693f6486caf5R80-R105))
*  Add the ToolsMinifyCssConfig type and the minifyCss property to the SharedToolsConfig interface in the builder-shared package, to allow users to customize the CssMinimizerPlugin options through the config.tools.minifyCss property ([link](https://github.com/web-infra-dev/modern.js/pull/4602/files?diff=unified&w=0#diff-981f6d7d7f0b83caea36c977fe36969dbc79c9944f45839916eceb480931b59cR17), [link](https://github.com/web-infra-dev/modern.js/pull/4602/files?diff=unified&w=0#diff-981f6d7d7f0b83caea36c977fe36969dbc79c9944f45839916eceb480931b59cR61-R62), [link](https://github.com/web-infra-dev/modern.js/pull/4602/files?diff=unified&w=0#diff-981f6d7d7f0b83caea36c977fe36969dbc79c9944f45839916eceb480931b59cR95-R99))
*  Remove the css-minimizer-webpack-plugin dependency from the builder-webpack-provider package, and use the applyCSSMinimizer function from the builder-shared package instead ([link](https://github.com/web-infra-dev/modern.js/pull/4602/files?diff=unified&w=0#diff-3497897bafd468150e9ba75f0d2843fcd2cc7995518584f7fa73e853a153f1a1L103), [link](https://github.com/web-infra-dev/modern.js/pull/4602/files?diff=unified&w=0#diff-f0900604339bc45e29308a86e37b7d2de6252c91d274db00cfc834db480a73dcL3-R9), [link](https://github.com/web-infra-dev/modern.js/pull/4602/files?diff=unified&w=0#diff-f0900604339bc45e29308a86e37b7d2de6252c91d274db00cfc834db480a73dcL28-R28))
*  Remove the ToolsMinifyCssConfig type and the minifyCss property from the ToolsConfig interface in the builder-webpack-provider package, since they are now inherited from the SharedToolsConfig interface in the builder-shared package ([link](https://github.com/web-infra-dev/modern.js/pull/4602/files?diff=unified&w=0#diff-220a653dee9c86fd41826422377e54d817b79fa77e5cd694d29fc0db6257ffc8L17), [link](https://github.com/web-infra-dev/modern.js/pull/4602/files?diff=unified&w=0#diff-220a653dee9c86fd41826422377e54d817b79fa77e5cd694d29fc0db6257ffc8L28-L29), [link](https://github.com/web-infra-dev/modern.js/pull/4602/files?diff=unified&w=0#diff-220a653dee9c86fd41826422377e54d817b79fa77e5cd694d29fc0db6257ffc8L80-L83))
*  Remove the unused imports and exports of the CssMinimizerPlugin and CssMinimizerPluginOptions types from the builder-webpack-provider and builder-shared packages ([link](https://github.com/web-infra-dev/modern.js/pull/4602/files?diff=unified&w=0#diff-9c109a48e5f7bdd5ab7ee9bf6e13f0d765b5e9bafd6d16c9ff514073ff1004e1L17-R21), [link](https://github.com/web-infra-dev/modern.js/pull/4602/files?diff=unified&w=0#diff-753e5e69851353e5879f3c314ec36e4b9df1b5848a2ff1cd087950d9583801b9L4), [link](https://github.com/web-infra-dev/modern.js/pull/4602/files?diff=unified&w=0#diff-753e5e69851353e5879f3c314ec36e4b9df1b5848a2ff1cd087950d9583801b9L13-L17))
*  Remove the tips and links that indicate the minifyCss config option and the image compress plugin are only supported by webpack from the builder-doc package, since they are now supported by both webpack and rspack bundlers ([link](https://github.com/web-infra-dev/modern.js/pull/4602/files?diff=unified&w=0#diff-6793f4bf97bd2e620d52ac92683554ab4427a4c09d4d9ad2b38647c4880c5a21L17-L18), [link](https://github.com/web-infra-dev/modern.js/pull/4602/files?diff=unified&w=0#diff-914057dea134889c31535df0ce07bab3b24e410174c7667d2ca98f2e7fef8118L119), [link](https://github.com/web-infra-dev/modern.js/pull/4602/files?diff=unified&w=0#diff-8f5310cfafd0351d4d820936d8fceedd959f94347054ad26444f7e84466cfc48L5-L8), [link](https://github.com/web-infra-dev/modern.js/pull/4602/files?diff=unified&w=0#diff-eb42633a00f3c2f9f57a18a8612bb50cedc3e1be57b193f25b33788129b84971L17-L18), [link](https://github.com/web-infra-dev/modern.js/pull/4602/files?diff=unified&w=0#diff-f8cbd3f0d3d051c8b2816d35026eb70aed638a236d5ded5ac7ce1d145db54a62L119), [link](https://github.com/web-infra-dev/modern.js/pull/4602/files?diff=unified&w=0#diff-c17a92994ced4b6e76cc3ddf96aa3e1b327f97172fa289a95e19424ca1ca90ebL5-L8))
*  Update the test cases for the css-minimize and css-modules-composes features in the builder package to expect the same CSS output for both webpack and rspack bundlers, since they both use the same CssMinimizerPlugin to minimize CSS files ([link](https://github.com/web-infra-dev/modern.js/pull/4602/files?diff=unified&w=0#diff-6efa7765ca989896b38d506a1414605052cb16d72c5a0794d224ee38eb1a1a81L16-R18), [link](https://github.com/web-infra-dev/modern.js/pull/4602/files?diff=unified&w=0#diff-2b29e21aed37a368ba6409d8bf483cd3bea8c90cd6474c1fe4e6a0da2d890578L15-R17), [link](https://github.com/web-infra-dev/modern.js/pull/4602/files?diff=unified&w=0#diff-2b29e21aed37a368ba6409d8bf483cd3bea8c90cd6474c1fe4e6a0da2d890578L36-R32))

## Checklist

- [ ] I have added changeset via `pnpm run change`.
- [ ] I have updated the documentation.
- [ ] I have added tests to cover my changes.

### CODE REVIEW

File: packages/builder/builder-rspack-provider/tests/plugins/minimize.test.ts [hunk 1]
```diff
@@ -107,16 +172,48 @@ describe('plugins/minimize', () => {
...
-        "pureFuncs": [
-          "console.log",
-          "console.warn",
-        ],
-      }
-    `);
+    expect(bundlerConfigs[0].optimization?.minimizer).toMatchInlineSnapshot(
+      `
+      [
+        Plugin {
```
Comment (ahabhgk) [inline packages/builder/builder-rspack-provider/tests/plugins/minimize.test.ts:178]: How can I make this be `SwcJsMinimizerPlugin`, is this read `[Symbol.toString]`?

Comment (9aoy) [reply packages/builder/builder-rspack-provider/tests/plugins/minimize.test.ts:178]: it seems get from constructor's name, maybe you can define a name for SwcJsMinimizerPlugin class?

Comment (9aoy) [reply packages/builder/builder-rspack-provider/tests/plugins/minimize.test.ts:178]: Disregard me, it's not constructor's name...

Comment (9aoy) [reply packages/builder/builder-rspack-provider/tests/plugins/minimize.test.ts:178]: Actually, I don't know, it seems to take the class name



but, if you add a instance name,this also can be output like this:

image

**Review by chenjiahan — APPROVED** [review]

**Pull request merged**